File: progs/javascript/eagerness.js
//Eagerness: eager
let counter = 0;

async function print(msg) {
  console.log(msg);
}

async function asyncInt() {
  let me = counter++;
  let until = Math.floor(Math.random() * 50);
  for (let i = 0; i < until; i++)
    await print(`[${me}] waiting`);
  return until;
}

async function main() {
  let a = asyncInt();
  let b = asyncInt();
  console.log("working");
  let c = await a + await b;
  await print(c);
}

main().then(() => console.log("exiting"))

File: progs/python/aio/eagerness.py
#Eagerness: Lazy
import asyncio
from random import randint

counter = 0

async def write(msg):
    print(msg)

async def async_int():
    global counter
    me = counter
    counter += 1
    until = randint(0, 50)
    for _ in range(until):
        await write(f"{me} waiting")
    return until

async def main():
    a = async_int()
    b = async_int()
    print("working")
    c = await a + await b
    await write(c)

asyncio.run(main())
